### PR TITLE
char value encoder

### PR DESCRIPTION
**Pull request by danfruehauf**

### CODE REVIEW

File: src/main/java/au/org/emii/ncdfgenerator/CharValueEncoder.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                fill = s.charAt(0);
+            }
+            catch (Exception e) {
+                throw new NcdfGeneratorException("Expected _FillValue attribute to be Char type");
+            }
+            haveFill = true;
+        }
+    }
+
+    public void encode(Array array, int ima, Object value) throws NcdfGeneratorException {
```
Comment (jonescc) [inline src/main/java/au/org/emii/ncdfgenerator/CharValueEncoder.java]: ima?

File: src/main/java/au/org/emii/ncdfgenerator/CharValueEncoder.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        if (attributes.get("_FillValue") != null) {
+            try {
+                String s = (String)attributes.get("_FillValue");
+                fill = s.charAt(0);
+            }
+            catch (Exception e) {
+                throw new NcdfGeneratorException("Expected _FillValue attribute to be Char type");
+            }
+            haveFill = true;
+        }
```
Comment (jonescc) [inline src/main/java/au/org/emii/ncdfgenerator/CharValueEncoder.java:37]: Should we be throwing an exception here, or is it OK to have no fillvalue if there are no missing values in which case this is not an exception

File: src/main/java/au/org/emii/ncdfgenerator/CharValueEncoder.java (hunk 1 above)
Comment (danfruehauf) [inline src/main/java/au/org/emii/ncdfgenerator/CharValueEncoder.java]: A bit of a copy/paste from the float encoder.

File: src/main/java/au/org/emii/ncdfgenerator/CharValueEncoder.java (hunk 2 above)
Comment (danfruehauf) [inline src/main/java/au/org/emii/ncdfgenerator/CharValueEncoder.java:37]: No fill - no big deal.

**Comment by danfruehauf** [comment]
@jonescc Ready again for review. This time it has an example in the README file which works.

**Pull request merged**